### PR TITLE
feat: [CDS-97931]: update doc regarding https module source for terra…

### PR DESCRIPTION
**Pull request by vlica-harness**
…form

Thanks for contributing to the Harness Developer Hub! Our code owners will review your submission.

## Description

* Please describe your changes: \__________________________________
* Jira/GitHub Issue numbers (if any): \______________________________
* Preview links/images (Internal contributors only): \__________________

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [ ] Successful preview build.
- [ ] Code owner review.
- [ ] No merge conflicts.
- [ ] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

**Comment by vlica-harness** [comment]
hi @krishi0408 @SushrutHarness @vishal-av 
Please review/approve the PR

File: docs/continuous-delivery/cd-infrastructure/terraform-infra/run-a-terraform-plan-with-the-terraform-apply-step.md [hunk 1]
```diff
@@ -201,7 +201,7 @@ In **Source Module**, you can select **Use Connector credentials** to have Harne
 
 If you do not select **Use Connector credentials**, Terraform will use the credentials that have been set up in the system.
 
-The **Use Connector credentials** setting is limited to Harness Git Connectors using SSH authentication (not HTTPS) and a token.
+The **Use Connector credentials** setting is available to Harness Git Connectors using SSH authentication either HTTPS.
```
Comment (SushrutHarness) [inline docs/continuous-delivery/cd-infrastructure/terraform-infra/run-a-terraform-plan-with-the-terraform-apply-step.md:204]:
```suggestion
The **Use Connector credentials** setting is available to Harness Git Connectors using SSH authentication or HTTPS.
```

File: docs/continuous-delivery/cd-infrastructure/terraform-infra/run-a-terraform-plan-with-the-terraform-plan-step.md [hunk 2]
```diff
@@ -197,7 +197,7 @@ In **Source Module**, you can select **Use Connector credentials** to have Harne
 
 If you do not select **Use Connector credentials**, Terraform will use the credentials that have been set up in the system.
 
-The **Use Connector credentials** setting is limited to Harness Git Connectors using SSH authentication (not HTTPS) and a token.
+The **Use Connector credentials** setting is available to Harness Git Connectors using SSH authentication either HTTPS.
```
Comment (SushrutHarness) [inline docs/continuous-delivery/cd-infrastructure/terraform-infra/run-a-terraform-plan-with-the-terraform-plan-step.md:200]:
```suggestion
The **Use Connector credentials** setting is available to Harness Git Connectors using SSH authentication or HTTPS.
```

**Review by SushrutHarness — CHANGES_REQUESTED** [review]
To clarify, is the following now true?:

Users will now see the "Use connector Credentials" setting with HTTPS Git Connectors.

If so, we can accept the suggestions and merge

**Review by SushrutHarness — APPROVED** [review]
LGTM

**Pull request merged**